packages/headless/src/test/mock-controller-definitions.ts

```diff
@@ -0,0 +1,109 @@
+import {vi} from 'vitest';
+import type {CoreEngine, CoreEngineNext} from '../app/engine.js';
+import type {SearchEngine} from '../app/search-engine/search-engine.js';
+import type {Controller} from '../controllers/controller/headless-controller.js';
+import type {SSRCommerceEngine} from '../ssr/commerce/factories/build-factory.js';
+import type {Kind} from '../ssr/commerce/types/kind.js';
+import type {
+  ControllerDefinitionWithoutProps,
+  ControllerDefinitionWithProps,
+} from '../ssr-next/common/types/controllers.js';
+
+interface MockController {
+  initialState?: Record<string, unknown>;
+}
+
+interface MockControllerDefinitionWithoutProps
+  extends ControllerDefinitionWithoutProps<SearchEngine, Controller> {}
+
+interface MockControllerDefinitionWithProps
+  extends ControllerDefinitionWithProps<
+    SearchEngine,
+    Controller,
+    MockController
+  > {}
+
+type SolutionTypeAvailabilities = {
+  listing?: boolean;
+  search?: boolean;
+  standalone?: boolean;
+  recommendation?: boolean;
+};
+
+export function buildMockController(): Controller {
+  return {
+    subscribe: vi.fn(),
+    state: {},
+  } as Controller;
+}
+
+export function buildMockControllerWithInitialState(
+  _engine: CoreEngine | CoreEngineNext | SSRCommerceEngine,
+  props: MockController
+): Controller {
+  return {
+    state: props.initialState,
+    subscribe: vi.fn(),
+  } as Controller;
+}
+
+/**
+ * For SSR use only.
```

Comment:
Isn't everything in this file pretty much intended for SSR use only? 😄 

Maybe we could rename the file to something like

`mock-ssr-controller-definitions.ts`

and get rid of these comments.

Reply:
I have split the mock file into 2